cmds/shutdown/shutdown.go

```diff
@@ -47,17 +65,44 @@ func main() {
 		op = flag.Args()[0]
 	}
 
-	f, ok := opcodes[op]
+	if op == "" {
+		if flags.h && flags.r {
+			return 0, errors.New("Invalid aruments")
+		} else if flags.h {
+			op = "halt"
+		} else {
+			op = "reboot"
+		}
+
+	} else {
+		if flags.h || flags.r {
+			return 0, errors.New("Invalid aruments")
+		}
+	}
+
+	c, ok := opcodes[op]
 	if !ok {
+		return 0, errors.New("Invalid operation")
+	}
+	return c, nil
+}
+
+func main() {
+	flag.Parse()
+
+	f, err := ascertainOperation()
+	if err != nil {
+		fmt.Println(err)
 		usage()
 	}
 
-	if *dryrun {
-		log.Printf("unix.Reboot(0x%x)", f)
+	if flags.dryrun || flags.k {
+		fmt.Printf("unix.Reboot(0x%x)\n", f)
 		os.Exit(0)
 	}
 
 	if err := unix.Reboot(f); err != nil {
-		log.Fatalf(err.Error())
+		fmt.Println(err)
```

Comment:
log.Fatalf?

Reply:
I guess I went outside the scope a little bit here by removing `log` in favour of `fmt` as the logging output is being prefixed with a timestamp while `flag` generated output about invalid flags or `--help` does not.

I will revert back to `log` and can create another patch regarding the consistency of the output.